test/statement_test.ts

```diff
@@ -30,9 +30,9 @@ describe('statements', () => {
   });
   it('translates try/catch', () => {
     expectTranslate('try {} catch(e) {} finally {}')
-        .to.equal(' try { } catch ( e ) { } finally { }');
+        .to.equal(' try { } catch ( e , e_stack ) { } finally { }');
```

Comment:
Yegor, if I understand correctly you're saying no change is needed in
ts2dart because we should already be catching Error in the original angular
code? What change needs to be made in that case?

On Fri, May 8, 2015 at 3:03 PM Yegor <notifications@github.com> wrote:

> In test/statement_test.ts
> <https://github.com/angular/ts2dart/pull/153#discussion_r29980592>:
>
> > @@ -30,9 +30,9 @@ describe('statements', () => {
> >    });
> >    it('translates try/catch', () => {
> >      expectTranslate('try {} catch(e) {} finally {}')
> > -        .to.equal(' try { } catch ( e ) { } finally { }');
> > +        .to.equal(' try { } catch ( e , e_stack ) { } finally { }');
>
> Dart, like JS, has Error, which has a getter stackTrace. We could,
> instead of catch(e, s), do this:
>
> } catch(Error e) {
>   print(e.stackTrace);
> }
>
> All Dart's own standard errors are subclasses of Error already, and so is
> our BaseException. So it should work just fine.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/angular/ts2dart/pull/153/files#r29980592>.
>